cli/src/test/java/com/devonfw/tools/ide/tool/jmc/JmcTest.java

```diff
@@ -79,24 +93,124 @@ private void setupMockServerResponse(String testUrl, String contentType, String
   }
 
   @Test
-  public void jmcPostInstallShouldMoveFilesIfRequired() throws IOException {
+  public void jmcPostInstallShouldMoveFilesIfRequiredMockedServer() throws IOException {
 
     // arrange
     String path = "workspaces/foo-test/my-git-repo";
-    IdeContext context = newContext("basic", path, true);
+    IdeContext context = newContext("jmc", path, true);
     InstallCommandlet install = context.getCommandletManager().getCommandlet(InstallCommandlet.class);
     install.tool.setValueAsString("jmc", context);
     mockWebServer();
     // act
     install.run();
 
     // assert
+    performPostInstallAssertion(context);
+  }
+
+  @Test
+  public void jmcPostInstallShouldMoveFilesIfRequired() {
+
+    // arrange
+    String path = "workspaces/foo-test/my-git-repo";
+    String projectTestCaseName = "jmc";
+
+    ToolRepositoryMock toolRepositoryMock = buildToolRepositoryMockForJmc(projectTestCaseName);
+
+    IdeContext context = newContext(projectTestCaseName, path, true, toolRepositoryMock);
+    toolRepositoryMock.setContext(context);
+
+    CommandLetExtractorMock commandLetExtractorMock = new CommandLetExtractorMock(context);
+    Jmc commandlet = new Jmc(context);
+    commandlet.setCommandletFileExtractor(commandLetExtractorMock);
+
+    // act
+    commandlet.install();
+
+    // assert
+    performPostInstallAssertion(context);
+  }
+
+  // TODO: This test currently fails because since JMC runs as a background process
+  // the test does not wait for the started process to finish and assertion is performed too early!
+  @Test
+  @Disabled
```

Comment:
When we run a process in the background, we cannot be sure that it finishes before the test continues to run. I did not consider this before merging the changes due to processMode usage.